gui/src/components/Taipy/Slider.tsx

```diff
@@ -195,46 +201,77 @@ const Slider = (props: SliderProps) => {
 
     useEffect(() => {
         if (props.value === undefined) {
-            let val = 0;
-            if (defaultValue !== undefined) {
-                if (typeof defaultValue === "string") {
-                    if (lovList.length) {
-                        try {
-                            const arrVal = JSON.parse(defaultValue) as string[];
-                            val = lovList.findIndex((item) => item.id === arrVal[0]);
-                            val = val === -1 ? 0 : val;
-                        } catch (e) {
-                            // Too bad also
-                        }
-                    } else {
-                        try {
-                            val = parseInt(defaultValue, 10);
-                        } catch (e) {
-                            // too bad
-                        }
-                    }
-                } else {
-                    val = defaultValue as number;
-                }
+            if (parsedDefaultValue !== undefined) {
+                setValue(parsedDefaultValue);
             }
-            setValue(val);
         } else {
             if (lovList.length) {
-                const val = lovList.findIndex((item) => item.id === props.value);
-                setValue(val === -1 ? 0 : val);
+                if (Array.isArray(props.value)) {
+                    setValue(props.value.map((i) => lovList.findIndex((j) => j.id === i))
+                        // Force unknown values to index 0
+                        .map((v) => v === -1 ? 0 : v))
+                }
+                else {
+                    const val = lovList.findIndex((item) => item.id === props.value);
+                    setValue(val === -1 ? 0 : val);
+                }
+            } else {
+                setValue(Array.isArray(props.value) ? props.value as number[] : props.value as number);
+            }
+        }
+    }, [props.value, lovList, parsedDefaultValue]);
+
+    // parsedDefaultValue init
+    const parseDefautValue = useCallback(() => {
```

Comment:
parsedDefaultValue could  be defined by a useMemo (instead of useState, useEffect, useCallback) ?

Reply:
Done. Thx